compiler/testData/diagnostics/tests/inference/expectedTypeAdditionalTest.fir.kt

```diff
@@ -7,7 +7,7 @@ fun <T> foo() = foo() as T
 
 fun <T> foo2(): T = TODO()
 
-val test = foo2().plus("") as String
+val test = foo2().plus("") <!USELESS_CAST!>as String<!>
```

Comment:
`foo2`'s return type is `T`, but again, with smartcast, its type is seen as `Nothing`. Not sure how `T/Nothing.plus` is resolved, and I think if that becomes unresolved or error type, this warning would be gone. Anyhow, with that resolution, `.plus("")` is `String`, and thus redundant cast. :\